setup.py

```diff
@@ -44,18 +52,85 @@
     frida_extension = os.environ['FRIDA_EXTENSION']
 frida_major_version = int(frida_version.split(".")[0])
 
+Tag = namedtuple("Tag", ["tagname", "attrs"])
+ParsedUrlInfo = namedtuple("ParsedUrlInfo",
+                           ["url", "filename", "major", "minor", "micro"])
+
+
+def get_index_url_from_pip(pip):
+    cmd = [pip, "config", "get", "global.index-url"]
```

Comment:
Could we do this through the API instead, or is `pip config` relying on private APIs?

Reply:
As far as I know, pip [doesn't provide any APIs](https://pip.pypa.io/en/stable/user_guide/#using-pip-from-your-program). The only interface it provides is the command line interface. The whole of pip's implementation is in its `_internal` package, which starts with an underline.

In addition, `[sys.executable, "-m", "pip", ...]` looks better than `["pip", ...]` here. (I forgot to read its doc yesterday.) In this way, `FRIDA_PIP` may become useless.

Reply:
Ok, makes sense, thanks for clarifying. (Sorry for the delay here btw!)